From 3e2a6080ca645c1fa668339575d49ce4fa31e6c4 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Tue, 23 Sep 2025 13:33:43 +0100
Subject: [PATCH 1/2] Downstream patch: always find the system-wide ruff
 executable
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The following issue is for uv, but the situation is exactly the same.

“Should uv.find_uv_bin() be able to find /usr/bin/uv?”
https://github.com/astral-sh/uv/issues/4451
---
 python/ruff/_find_ruff.py | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/python/ruff/_find_ruff.py b/python/ruff/_find_ruff.py
index c0213bb23f..92a9486483 100644
--- a/python/ruff/_find_ruff.py
+++ b/python/ruff/_find_ruff.py
@@ -38,6 +38,16 @@ def find_ruff_bin() -> str:
         sysconfig.get_path("scripts", scheme=_user_scheme()),
     ]
 
+    # Downstream patch: always find the system-wide ruff executable
+    #
+    # The following issue is for uv, but the situation is exactly the same.
+    #
+    # “Should uv.find_uv_bin() be able to find /usr/bin/uv?”
+    #  https://github.com/astral-sh/uv/issues/4451
+    bindir_path = sysconfig.get_config_var("BINDIR")
+    if bindir_path is not None:
+        targets.insert(0, bindir_path)
+
     seen = []
     for target in targets:
         if not target:
-- 
2.55.0

